source/rust_verify/src/rust_to_vir_expr.rs

```diff
@@ -2461,6 +2461,53 @@ fn expr_assign_to_vir_innermost<'tcx>(
         Some(op) => Some(binopkind_to_binaryop(bctx, op, tc, lhs, rhs, mode_for_ghostness)?),
         None => None,
     };
+
+    // NOTE: A temparary solution for index_mut until mutable reference support lands.
+    match &lhs.kind {
+        ExprKind::Index(tgt_expr, idx_expr, _span) => {
+            let tgt_modifier = is_expr_typ_mut_ref(
+                bctx.types.expr_ty_adjusted(&tgt_expr),
+                ExprModifier::ADDR_OF_MUT,
+            )?;
+            let tgt_vir = expr_to_vir(bctx, tgt_expr, tgt_modifier)?;
+            let idx_vir = expr_to_vir(bctx, idx_expr, ExprModifier::REGULAR)?;
+            let mut rhs_vir = expr_to_vir(bctx, rhs, modifier)?;
+            let fun = vir::fun!["vstd" => "std_specs", "core", "index_set"];
+            let typ_args = Some(Arc::new(vec![
+                undecorate_typ(&tgt_vir.typ),
+                idx_vir.typ.clone(),
+                rhs_vir.typ.clone(),
+            ]));
+            let tgt_vir =
+                bctx.spanned_typed_new(tgt_expr.span, &tgt_vir.typ.clone(), ExprX::Loc(tgt_vir));
+            let call_target = CallTarget::Fun(
+                vir::ast::CallTargetKind::Static,
+                fun,
+                typ_args.unwrap(),
+                Arc::new(vec![]),
+                AutospecUsage::Final,
+            );
+
+            if let Some(op) = op {
```

Comment:
This duplicates `lhs`, which would semantically correspond to evaluating `lhs` twice.  In Rust, `lhs` is only evaluated once.  I would recommend either not handling `Some(op)` for now, or if you really need it, limiting it to the case where `tgt` and `idx` are variables or constants, which should be semantically equivalent for array/slice/Vec.

Reply:
Thanks! To avoid the side effects, I checked can_have_side_effects for tgt and idx that would ensure we cannot use tgt/idx twice if they have side effects. In addition, I also checked exec mode and is_smt_arith for lhs and rhs pair.